src/briefcase/commands/create.py

```diff
@@ -568,13 +573,22 @@ def install_image(self, role, variant, size, source, target):
                 # Copy the source image to the target location
                 self.shutil.copy(full_source, target)
             else:
-                print(
+
+                source_suffix = source_filename[-5:]
+                if source_suffix == '.icns' and full_role == 'application icon':
```

Comment:
I'd be inclined to put this logic (including the "can't find variant" logic) into `create_variant` - That way the API you've got is "create a .icns application icon variant of size X", which will fail if that variant can't be satisfied. What you've effectively specified here is "Create a variant, but only if it's is one of the specifically approved types". 

Why does this matter? Think longer term: v1 will add support for at least one variant creation type. In the version of the code you've got here, you'll need to make 2 change - add the variant creation code, and then modify the logic so that the request to make the variant will be made at all. If all the logic is in create variant, you're just expanding the capabilities of create_variant so it won't return an error/warning in as many situations.

Reply:
I have moved this into create_image_variant in my upcoming commit. Thanks!